vfox-restServer/src/main/java/com/vfoexchange/restServer/controller/UserController.java

```diff
@@ -1,36 +1,182 @@
+import com.vfoexchange.restServer.dto.*;
+import com.vfoexchange.restServer.model.Services;
+import com.vfoexchange.restServer.service.EmailServices;
+import com.vfoexchange.restServer.service.UserService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.env.Environment;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 
-import com.vfoexchange.restServer.model.User;
-import com.vfoexchange.restServer.service.UserService;
-import com.vfoexchange.restServer.dto.UserDTO;
+import java.util.List;
 
 @RestController
 public class UserController {
 
     @Autowired
-    UserService userService;
+    private UserService userService;
+    @Autowired
+    private EmailServices emailServices;
+
+    @Autowired
+    Environment environment;
 
+    private static Logger LOGGER = LoggerFactory.getLogger(UserController.class);
     /*
-    Method for adding new user(as of now only for adviser, later should be moved to generic one)
+    Method for adding new user(user can be advisor or admin)
      */
-    @RequestMapping(value = "/add/user", method = RequestMethod.POST)
+    @RequestMapping(value = "/add/user", method = RequestMethod.POST, consumes = MediaType.APPLICATION_JSON_VALUE, produces = MediaType.APPLICATION_JSON_VALUE)
+    @ResponseBody
+    public ResponseEntity<ResponseDTO> addUser(@RequestBody UserDTO userDto) {
+
+        ResponseDTO resp = new ResponseDTO();
+        ResponseEntity<ResponseDTO> responseEntity;
+        if (userService.isAleadyExist(userDto.getUsername())) {
+            resp.setCode(HttpStatus.ALREADY_REPORTED.toString());
+            resp.setMsg("This email already exists in our system. Please try another email for registration");
+            responseEntity = new ResponseEntity<ResponseDTO>(resp, HttpStatus.OK);
+            LOGGER.info(userDto.getUsername()+" email already exists in our system.");
+            return responseEntity;
+        }
+        try {
+
+            emailServices.sendMail(userDto.getUsername());
+            userService.addUser(userDto);
+            resp.setCode(HttpStatus.OK.toString());
+            resp.setMsg("Your account has been created,  please verify it by clicking the activation link that has been send to your email.");
+            responseEntity = new ResponseEntity<ResponseDTO>(resp, HttpStatus.OK);
+            LOGGER.info("Mail has been sent for verify");
+        } catch (Exception e) {
+            LOGGER.error("User " + e.getMessage());
+            resp.setCode(HttpStatus.BAD_REQUEST.toString());
+            resp.setMsg("Error occurred while sending mail.");
+            responseEntity = new ResponseEntity<ResponseDTO>(resp, HttpStatus.OK);
+        }
+
+        return responseEntity;
+    }
+
+
```

Comment:
Remove unnecessary blank line.

Reply:
Unnecessary blank line removed and committed.